### PR TITLE
Merge release v1.1.2 into dev

### PR DESCRIPTION
**Pull request by jboddey**

### CODE REVIEW

**Comment by jboddey** [comment]
Requires merge conflicts in UI

**Comment by jboddey** [comment]
@sofyakurilova could you resolve the merge conflicts within the UI code for this PR?

File: modules/ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -20,21 +20,33 @@ const routes: Routes = [
   {
     path: 'testrun',
     loadChildren: () =>
+<<<<<<< HEAD
+      import('./progress/progress.module').then(m => m.ProgressModule),
+=======
       import('./pages/testrun/progress.module').then(m => m.ProgressModule),
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app-routing.module.ts:27]: @sofyakurilova,
Please check how the conflicts are resolved.
We should re-work it.

File: modules/ui/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -20,21 +20,33 @@ const routes: Routes = [
...
     title: 'Testrun',
   },
   {
     path: 'devices',
     loadChildren: () =>
+<<<<<<< HEAD
+      import('./device-repository/device-repository.module').then(
+=======
       import('./pages/devices/device-repository.module').then(
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app-routing.module.ts:37]: Please resolve the conflicts.

File: modules/ui/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -20,21 +20,33 @@ const routes: Routes = [
...
     title: 'Testrun - Devices',
   },
   {
     path: 'reports',
     loadChildren: () =>
+<<<<<<< HEAD
+      import('./history/history.module').then(m => m.HistoryModule),
+=======
       import('./pages/reports/history.module').then(m => m.HistoryModule),
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app-routing.module.ts:49]: The same: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.html [hunk 4]
```diff
@@ -14,11 +14,15 @@
...
 <app-bypass></app-bypass>
+<<<<<<< HEAD
+<mat-drawer-container hasBackdrop="false" class="app-container" autosize>
+=======
 <mat-drawer-container
   [class.active-menu]="isMenuOpen | async"
   hasBackdrop="false"
   class="app-container"
   autosize>
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.html:25]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.html [hunk 5]
```diff
@@ -103,6 +114,7 @@ <h1 class="main-heading">Testrun</h1>
       <app-callout
         [type]="CalloutType.Info"
         *ngIf="(hasConnectionSetting$ | async) === false">
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.html:117]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.html [hunk 6]
```diff
@@ -119,7 +131,12 @@ <h1 class="main-heading">Testrun</h1>
         [type]="CalloutType.Info"
         *ngIf="
           (hasConnectionSetting$ | async) === true &&
+<<<<<<< HEAD
+          !(devices$ | async)?.length &&
+          isDevicesLoaded
+=======
           !(devices$ | async)?.length
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.html:139]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.html [hunk 7]
```diff
@@ -141,7 +158,11 @@ <h1 class="main-heading">Testrun</h1>
           (devices$ | async)?.length &&
           (!(systemStatus$ | async)?.status ||
             (systemStatus$ | async)?.status === StatusOfTestrun.Idle) &&
+<<<<<<< HEAD
+          isStatusLoaded &&
+=======
           isStatusLoaded === true &&
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.html:165]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.html [hunk 8]
```diff
@@ -165,7 +186,14 @@ <h1 class="main-heading">Testrun</h1>
...
     position="end"
     class="settings-drawer">
+<<<<<<< HEAD
+    <app-general-settings
+      [interfaces]="interfaces"
+      (closeSettingEvent)="closeSetting()"
+      (reloadInterfacesEvent)="reloadInterfaces()">
+=======
     <app-general-settings (closeSettingEvent)="closeSetting()">
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.html:196]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.scss [hunk 9]
```diff
@@ -40,7 +40,11 @@ $nav-open-btn-width: 210px;
   margin-top: $toolbar-height;
 }
 
+<<<<<<< HEAD
+:host.active-menu {
+=======
 .active-menu {
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.scss:47]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 10]
```diff
@@ -43,23 +43,34 @@ import {
...
+import { StateService } from './services/state.service';
+=======
 import { Routes } from './model/routes';
 import { MockStore, provideMockStore } from '@ngrx/store/testing';
 import { State } from '@ngrx/store';
 import { appFeatureKey } from './store/reducers';
 import { FocusManagerService } from './services/focus-manager.service';
 import { AppState } from './store/state';
 import { toggleMenu, updateFocusNavigation } from './store/actions';
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:58]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 11]
```diff
@@ -43,23 +43,34 @@ import {
...
   let router: Router;
   let mockService: SpyObj<TestRunService>;
+<<<<<<< HEAD
+  let mockLoaderService: SpyObj<LoaderService>;
+  let mockStateService: SpyObj<StateService>;
+=======
   let store: MockStore<AppState>;
   let focusNavigation = true;
   let mockFocusManagerService: SpyObj<FocusManagerService>;
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:73]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 12]
```diff
@@ -84,18 +95,35 @@ describe('AppComponent', () => {
...
+    mockLoaderService = jasmine.createSpyObj(['setLoading']);
+    mockStateService = jasmine.createSpyObj('mockStateService', [
+      'focusFirstElementInMain',
+=======
       'setIsOpenStartTestrun',
     ]);
 
     mockFocusManagerService = jasmine.createSpyObj('mockFocusManagerService', [
       'focusFirstElementInContainer',
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:112]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 13]
```diff
@@ -84,18 +95,35 @@ describe('AppComponent', () => {
...
     );
+<<<<<<< HEAD
+    mockService.getSystemInterfaces.and.returnValue(of({}));
     (mockService.systemStatus$ as unknown) = of({});
     mockService.isTestrunStarted$ = of(true);
+    mockService.hasConnectionSetting$ = of(true);
+=======
+    (mockService.systemStatus$ as unknown) = of({});
+    mockService.isTestrunStarted$ = of(true);
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:126]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 14]
```diff
@@ -126,6 +158,7 @@ describe('AppComponent', () => {
         },
         provideMockStore({}),
         { provide: FocusManagerService, useValue: mockFocusManagerService },
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:161]: The same here: Please check how the conflicts are resolved.

File: modules/ui/src/app/app.component.spec.ts [hunk 15]
```diff
@@ -249,7 +282,11 @@ describe('AppComponent', () => {
     });
   }));
 
+<<<<<<< HEAD
+  it('should call focusFirstElementInMain if settingsDrawer opened not from toggleBtn', fakeAsync(() => {
+=======
   it('should call focusFirstElementInContainer if settingsDrawer opened not from toggleBtn', fakeAsync(() => {
+>>>>>>> dev
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/app.component.spec.ts:289]: The same here: Please check how the conflicts are resolved.

**Review by OlgaMardvilko — CHANGES_REQUESTED** [review]
Please check how the conflicts are resolved.
There are a lot of issues related to resolving conflicts. 
It should be re-worked.

File: modules/ui/src/app/services/state.service.ts [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+    const firstControl: HTMLElement | null = window.document.querySelector(
+      '#main button:not([disabled="true"]), ' +
+        '#main a:not([disabled="true"]), #main table'
+    );
+
+    if (firstControl) {
+      firstControl.focus();
+    }
+  }
+}
```
Comment (OlgaMardvilko) [inline modules/ui/src/app/services/state.service.ts:17]: I don't think we need to push this changes to dev as we didn't do anything related to this on the release v1.1.2. So, the dev branch has more correct last changes.
Let discuss.

**Review by OlgaMardvilko — CHANGES_REQUESTED** [review]

**Review by OlgaMardvilko — APPROVED** [review]

**Pull request merged**